app/src/main/java/com/github/se/studybuddies/database/DatabaseConnection.kt

```diff
@@ -213,6 +229,16 @@ class DatabaseConnection {
         }
   }
 
+  fun updateLocation(uid: String, location: String) {
```

Comment:
There are a few places (like here) when you need the location as a string. Given that the two possible states of the location are "online" or "offline", maybe create an enum class LocationState or sth and have it contain OFFLINE and ONLINE. This way, when you need to make checks, you can just have a switch statement over the types of LocationState and there can't be any bug because of mistyping

Reply:
Thanks for your comment but maybe the code wasn't clear enough or you misunderstood something. But there aren't just 2 states "offline" and "online" for the location. What I thought was to set the location to "offline" when the user isn't tracking its location and he closes the app. On the other hand, we the user is using the app, the location it set to its current location. I give this location as string in updateLocation because it was easier. Thus the location should either be "offline" or a location (for example "40.0034,78.02323"). I hope this was clear. Do you have any other suggestions ?

Reply:
oooh okay I understand! In that case maybe just extract the string like the others to avoid hard-coding it and all good !